src/CSET/recipes/__init__.py

```diff
@@ -64,25 +85,24 @@ def unpack_recipes(recipe_dir: Path, recipe_name: str = None) -> None:
         If recipe_dir cannot be created, such as insufficient permissions, or
         lack of space.
     """
-    logging.debug("Saving recipes to %s", recipe_dir)
+    file = _get_recipe_file(recipe_name)
     recipe_dir.mkdir(parents=True, exist_ok=True)
-    for file in _recipe_files_in_tree(recipe_name):
-        logging.debug("Saving %s", file.name)
-        if recipe_dir.joinpath(file.name).exists():
-            warnings.warn(
-                f"{file.name} already exists in target directory, skipping.",
-                stacklevel=2,
-            )
-        else:
-            logging.info("Unpacking %s to %s", file.name, recipe_dir / file.name)
-            (recipe_dir / file.name).write_bytes(file.read_bytes())
+    output_file = recipe_dir / file.name
+    logging.debug("Saving recipe to %s", output_file)
+    if output_file.exists():
+        warnings.warn(
+            f"{file.name} already exists in target directory, skipping.", stacklevel=2
+        )
+        return
+    logging.info("Unpacking %s to %s", file.name, output_file)
```

Comment:
I've done it this way in all calls to logging, as logging does its substituion lazily. That is, if your log level wouldn't show that message, it doesn't bother to construct the string. If I was to use an f-string instead it would contruct the string before deciding whether it needed it.

In reality constructing a string is pretty cheap, and this would only matter inside of hot loops, but I've done it this way for consistency.